.github/workflows/ci-tests.yml

```diff
@@ -0,0 +1,104 @@
+# reference:
+#   - https://github.com/actions/cache
+#   - https://github.com/actions/checkout
+
+name: ci-tests
+
+on:
+  pull_request:
+
+  push:
+    branches:
+      - "main"
+      - "v*x"
+      - "!auto-update-lockfiles"
+      - "!pre-commit-ci-update-config"
+      - "!dependabot/*"
+    tags:
+      - "v*"
+
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+
+
+jobs:
+  tests:
+    name: "${{ matrix.session }} (py${{ matrix.python-version }} ${{ matrix.os }})"
+
+    runs-on: ${{ matrix.os }}
+
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    env:
+      ENV_NAME: "ci-tests"
+      CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+      #: If you change the IRIS_SOURCE here you will also need to change it in
+      #: the noxfile and the benchmark and wheel workflows.
+      IRIS_SOURCE: "github:main"
+      IRIS_TEST_DATA_LOC_PATH: tests
+      IRIS_TEST_DATA_PATH: tests/iris-test-data
+      IRIS_TEST_DATA_VERSION: "2.19"
+      ENV_CACHE_BUILD: "0"
+      TEST_DATA_CACHE_BUILD: "0"
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest"]
+        python-version: ["3.9", "3.10"]
+        include:
+          - python-version: "3.10"
+            coverage: true
+
+    steps:
+      - name: "checkout"
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```

Comment:
Unnecessary for the tests (uses a lot more resource)

Reply:
see https://github.com/SciTools-incubator/iris-esmf-regrid/commit/bb271a94b100620fe063aeffdb9d5951fd6b41d9